### ISSUE TITLE
Implement cross-validation approach

### ISSUE THREAD

**Issue by sgreenbury** [alan-turing-institute/autoemulate#344]

**Comment by radka-j** [comment]
The goal is to implement a simple KFold cross validation.

The [sklear.model_selection.KFold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html) object returns train/val split indices for each fold. What we need to implement is the `cross_validate` loop over the folds. So something like:

```python
from sklearn.model_selection import KFold
from torch.utils.data import Subset

val_scores = []
kfold = KFold(n_splits=k, shuffle=True)
for (train_idx, val_idx) in kfold.split(dataset):
    train_fold = Subset(dataset, train_idx)
    val_fold = Subset(dataset, val_idx) 

    # fit and evaluate model here
    ...
    val_scores.append(...)
```

**Comment by radka-j** [comment]
For a single train/validation data splits we can use [ShuffleSplit](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html).

**Comment by radka-j** [comment]
See also notes in #371 expanding on this.
